run/markdown-preview/editor/render.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const gcpMetadata = require('gcp-metadata')
+const got = require('got');
+
+// renderRequest creates a new HTTP request with IAM ID Token credential.
+// This token is automatically handled by private Cloud Run (fully managed) and Cloud Functions.
+const renderRequest = async (service, markdown) => { 
+  // [START run_secure_request]
+  let token;
+
+  // Build the request to the Renderer receiving service.
+  const serviceRequestOptions = { 
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    json: {markdown},
+    responseType: 'json'
```

Comment:
Your render service is set up to receive JSON, but why not just send it as plain text?

Reply:
Ultimately, I don't know how to set up the request to correctly pass the data as plain text, convert it, and return it. Can we walk through it together?